tools/StaticAnalysis/Exceptions/AzureRM.Compute/BreakingChangeIssues.csv

```diff
@@ -21,3 +21,75 @@
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.UpdateAzureRmVmssVM","Update-AzureRmVmssVM","0","3000","The type of property 'CreateOption' of type 'Microsoft.Azure.Commands.Compute.Models.PSVirtualMachineDataDisk' has changed from 'System.String' to 'Microsoft.Azure.Management.Compute.Models.DiskCreateOptionTypes'.","Change the type of property 'CreateOption' back to 'System.String'."
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.NewAzureRmVmssIpConfigCommand","New-AzureRmVmssIpConfig","0","3000","The type of property 'PrivateIPAddressVersion' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIPConfiguration' has changed from 'System.String' to 'System.Nullable`1[Microsoft.Azure.Management.Compute.Models.IPVersion]'.","Change the type of property 'PrivateIPAddressVersion' back to 'System.String'."
 "Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.RemoveAzureRmContainerService","Remove-AzureRmContainerService","0","1020","The cmdlet 'Remove-AzureRmContainerService' no longer has output type 'System.Void'.","Make cmdlet 'Remove-AzureRmContainerService' return type 'System.Void'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.Automation.InvokeAzureRmVMRunCommand","Invoke-AzureRmVMRunCommand","0","1050","The parameter set 'VMParameter' for cmdlet 'Invoke-AzureRmVMRunCommand' has been removed.","Add parameter set 'VMParameter' back to cmdlet 'Invoke-AzureRmVMRunCommand'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.RemoveAzureRmVmssDiagnosticsExtension","Remove-AzureRmVmssDiagnosticsExtension","0","1020","The cmdlet 'Remove-AzureRmVmssDiagnosticsExtension' no longer has output type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSet'.","Make cmdlet 'Remove-AzureRmVmssDiagnosticsExtension' return type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSet'."
+"Microsoft.Azure.Commands.Compute.dll","Microsoft.Azure.Commands.Compute.RemoveAzureRmVmssDiagnosticsExtension","Remove-AzureRmVmssDiagnosticsExtension","0","3010","The property 'IdentityIds' of type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity' has been removed.","Add the property 'IdentityIds' back to type 'Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity'."
```

Comment:
Based on this line, it seems `Microsoft.Azure.Management.Compute.Models.VirtualMachineScaleSetIdentity` is still being exposed as part of the PowerShell interface in some why. Otherwise, it wouldn't care if the model's public interface has changed.

Reply:
This is because the type of a parameter is VirtualMachineScaleSet.  Can I change it (the type of input parameter of a cmdlet) to PSVirtualMachineScaleSet type?

Reply:
Yes. You should use PSVirtualMachineScaleSet where ever you were using VirtualMachineScaleSet previously. Otherwise, it will cause breaking changes.